unit-tests/run-unit-tests.py

```diff
@@ -160,6 +160,34 @@ def cat( filename ):
         for line in remove_newlines( file ):
             out( line )
 
+def check_log(log):
+    # Normal logs are expected to have in last line:
+    #     "All tests passed (11 assertions in 1 test case)"
+    # Tests that have failures, however, will show:
+    #     "test cases: 1 | 1 failed
+    #      assertions: 9 | 6 passed | 3 failed"
+    for ctx in grep( r'^test cases:\s*(\d+) \|\s*(\d+) (passed|failed)', log ):
+        m = ctx['match']
+        total = int(m.group(1))
+        passed = int(m.group(2))
+        if m.group(3) == 'failed':
+            # "test cases: 1 | 1 failed"
+            passed = total - passed
+        if passed < total:
+            if total == 1  or  passed == 0:
+                desc = 'failed'
+            else:
+                desc = str(total - passed) + ' of ' + str(total) + ' failed'
+
+            if verbose:
+                error( red + testname + reset + ': ' + desc )
```

Comment:
You're using global variables from within a function... I think it'll work in this case, but is not good practice in general. Can we pass the context in via the arguments?

Reply:
I added the testname as an argument to the function. That is the only global variable I could see